include/ginkgo/core/base/device_matrix_data.hpp

```diff
@@ -146,6 +146,13 @@ class device_matrix_data {
      */
     void remove_zeros();
 
+    /**
+     * Sums up all duplicate entries pointing to the same non-zero location.
+     * The output will be sorted in row-major order, and it will only reallocate
+     * if duplicates exist.
+     */
+    void sum_duplicates();
```

Comment:
In the assembly contexts I had in mind, the data would not be sorted, because each element consists of a contiguous range describing the local submatrix.